cedar-policy/src/api.rs

```diff
@@ -1595,6 +1595,15 @@ pub enum PolicySetError {
     /// Expected a template, but a static policy was provided.
     #[error("expected a template, but a static policy was provided")]
     ExpectedTemplate,
+    /// Error when removing a policy
+    #[error("unable to remove static policy: {0}")]
+    RemovePolicyError(PolicyId),
+    /// Error when removing a template
+    #[error("unable to remove policy template: {0}")]
+    RemoveTemplateError(PolicyId),
+    /// Error when unlinking a template
+    #[error("unable to unlink policy template: {0}")]
+    UnlinkingError(PolicyId),
```

Comment:
do we want more detail in these errors about _why_ the operation failed? like, "because it does not exist"?